config/routes.rb

```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  get 'users/', to: 'users#index'
+  get 'users/:user_id/posts/:id', to: 'users#post'
+  get 'users/:user_id/posts', to: 'users#posts'
+  get 'users/:id', to: 'users#user'
```

Comment:
- [ ] the way that creating routes it's more verbose you can use `resources` it is much cleaner and you can only specify the `index` and `show` action  like this: 
```RUBY
resources :users, only: %i[index show] do 
    resources :posts, only: %i[index show]
end